pkg/proxy/config.go

```diff
@@ -88,10 +93,15 @@ func NewConfig(opts *proxyoptions.Options) (*Config, error) {
 		return nil, err
 	}
 
-	c.ShardsConfig, err = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(&clientcmd.ClientConfigLoadingRules{ExplicitPath: c.Options.ShardsKubeconfig}, nil).ClientConfig()
+	c.ShardsConfig, err = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: c.Options.ShardsKubeconfig},
+		// We override the Server here so that the user doesn't have to specify unused server value
+		// The Server must have HTTPS scheme otherwise CA won't be loaded (see IsConfigTransportTLS method)
+		&clientcmd.ConfigOverrides{ClusterInfo: clientcmdapi.Cluster{Server: "https://fakeserver.io"}}).ClientConfig()
```

Comment:
The fakeserver overrided here looks strange, is that needed, or can we just use whatever is specified in the shard kubeconfig?

Reply:
Oh I see now that the server is empty in the file, if it can't remain empty here would it perhaps be better to default it to the root-shard URL?

Reply:
yes, please see https://github.com/kcp-dev/kcp/pull/2382#discussion_r1026411277 for alternatives.

Reply:
But even if we stick with the kubeconfig approach instead of the separate flags (which personally I think is fine, and probably easier if folks already have a cert based root-kubeconfig that allows access to all shards), we could switch the fakeserver.io to the root shard URL, right?

Reply:
sure, we could do that. I think we could do that just to read the new config and then erase the host value.

Reply:
I'd like to hear one more opinion on it. I'm not convinced we should change it to use the root shard URL.

Reply:
I hope that's okay with you :)